contractcourt/chain_watcher.go

```diff
@@ -403,16 +404,41 @@ func (c *chainWatcher) closeObserver(spendNtfn *chainntnfs.SpendEvent) {
 
 			// If we are lucky, the remote peer sent us the correct
 			// commitment point during channel sync, such that we
-			// can sweep our funds.
-			// TODO(halseth): must handle the case where we haven't
-			// yet processed the chan sync message.
-			commitPoint, err := c.cfg.chanState.DataLossCommitPoint()
-			if err != nil {
-				log.Errorf("Unable to retrieve commitment "+
-					"point for channel(%v) with lost "+
-					"state: %v",
-					c.cfg.chanState.FundingOutpoint, err)
-				return
+			// can sweep our funds. If we cannot find the commit
+			// point, there's not much we can do other than wait
+			// for us to retrieve it. We will attempt to retrieve
+			// it from the peer each time we connect to it.
+			// TODO(halseth): actively initiate re-connection to
```

Comment:
This can eventually be hooked up into the channel life cycle management stuff. So to request a notification once a new node connects. Actually we already have this for use in the funding manager, so perhaps it can be used here?

Reply:
yeah i think we can use `NotifyWhenOnline` to just try if they ever reconnect

Reply:
I actually like the current polling more, since we might need some time to process the `ChannelReestablish` after they reconnect. In that case the first lookup after they connect might fail, so we have to wait a little bit anyway. 

If we don't wait before retrying at that point, we risk the peer coming online send us the `ChannelReestablish` and disconnects, and we will stall here waiting for them to connect again.